src/node/stage/approved.rs

```diff
@@ -1004,15 +1016,21 @@ impl Approved {
             return;
         }
 
-        if let (Some(addr), _) = self
+        if let Some(info) = self
             .shared_state
             .remove_member(&pub_id, core.network_params.recommended_section_size)
         {
             info!("handle Offline: {}", pub_id);
 
             self.members_changed = true;
-            core.transport.disconnect(addr);
+
+            core.transport.disconnect(*info.p2p_node.peer_addr());
             let _ = self.members_knowledge.remove(pub_id.name());
+
+            core.send_event(Event::MemberLeft {
```

Comment:
do we need to check whether core is elder, before broadcasting this left event out?

Reply:
Good catch :+1: